tracer/test/Datadog.Trace.ClrProfiler.Managed.Tests/AutoInstrumentation/AWS/MemoryStreamCarrierGetter.cs

```diff
@@ -0,0 +1,52 @@
+// <copyright file="MemoryStreamCarrierGetter.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Amazon.SimpleNotificationService.Model;
+using Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.SNSTests;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.Propagators;
+using Newtonsoft.Json;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.SNSTests
+{
+    public struct MemoryStreamCarrierGetter : ICarrierGetter<PublishRequestProxy>
+    {
+        private const string DatadogKey = "_datadog";
+
+        public IEnumerable<string> Get(PublishRequestProxy carrier, string key)
+        {
+            object attributeValue;
+
+            if (carrier.MessageAttributes.TryGetValue(DatadogKey, out attributeValue))
+            {
+                var messageAttributeValue = attributeValue as MessageAttributeValue;
+
+                if (messageAttributeValue != null && messageAttributeValue.BinaryValue is MemoryStream memoryStream)
+                {
+                    memoryStream.Position = 0;
+                    using var reader = new StreamReader(memoryStream, encoding: Encoding.UTF8, detectEncodingFromByteOrderMarks: true, bufferSize: -1, leaveOpen: true);
```

Comment:
I led you astray 😞 The buffer size apparently needs to be a positive number for some platforms.

```suggestion
                    using var reader = new StreamReader(memoryStream, encoding: Encoding.UTF8, detectEncodingFromByteOrderMarks: true, bufferSize: 128, leaveOpen: true);
```

Reply:
np nifty ci feature helps to point that out yesterday but I forgot to add the commit